bevy_matchbox/src/socket.rs

```diff
@@ -77,8 +95,8 @@ impl<C: BuildablePlurality> From<WebRtcSocketBuilder<C>> for MatchboxSocket<C> {
 impl<C: BuildablePlurality> From<(WebRtcSocket<C>, MessageLoopFuture)> for MatchboxSocket<C> {
     fn from((socket, message_loop_fut): (WebRtcSocket<C>, MessageLoopFuture)) -> Self {
         let task_pool = IoTaskPool::get();
-        task_pool.spawn(message_loop_fut).detach();
-        MatchboxSocket(socket)
+        let task = task_pool.spawn(message_loop_fut);
+        MatchboxSocket(socket, Box::new(task))
```

Comment:
Is the purpose only to make sure it holds the task, such that if the `MatchboxSocket` drops, the task gets dropped/cancelled?

Reply:
Exactly that